renderdoc/android/android.cpp

```diff
@@ -405,7 +405,14 @@ bool InstallRenderDocServer(const std::string &deviceID)
           "%s missing - ensure you build all ABIs your device can support for full compatibility",
           apk.c_str());
 
-    adbExecCommand(deviceID, "install -r -g \"" + apk + "\"");
+    Process::ProcessResult adbCheck = adbExecCommand(deviceID, "install -r -g \"" + apk + "\"");
+
+    if(!adbCheck.strStderror.empty())
+    {
+      RDCLOG("Failed to install APK. stderr: %s", adbCheck.strStderror.c_str());
+      RDCLOG("Retrying...");
+      adbExecCommand(deviceID, "install -r \"" + apk + "\"");
```

Comment:
If this doesn't work what are the consequences for the user? At the moment there's no indication that this failure happened and no documentation to suggest that they'll need to perform any manual steps. We should find a way to alert the user that this failure happened.

That said, I'm looking at the permissions we currently require: `WRITE_EXTERNAL_STORAGE` and `INTERNET`. I *believe* that `WRITE_EXTERNAL_STORAGE` is no longer needed, as of #739, and that `INTERNET` is not needed either since we use abstract sockets for all communication, as of #822. Although I'm not sure if the permission came _back_ again for something.

Does this maybe mean we can drop the permissions requirements and drop the `-g` parameter as well?

Reply:
Currently, the capture is saved to the app-private external storage (/sdcard/Android/data/[packagename]/) so we do need the `WRITE_EXTERNAL_STORAGE` permission.
(We wouldn't need this permission if we saved it to the internal storage (/data/data/[packagename]/) but that's not good for us because only the app can access it's internal storage and we couldn't replay the capture.)

`INTERNET` permission is a normal level permission, the system automatically grants it at install time (we don't need to take care of it). 

If the `WRITE_EXTERNAL_STORAGE` permission isn't granted at install time, there will be a dialog window asking the user to grant it. It's familiar for the Android users, I don't think we need to give any further notification. (screenshot attached)

![screenshot_20181026-112048](https://user-images.githubusercontent.com/36237118/47562180-ff947b00-d91d-11e8-88fa-be2c37118e79.png)

Reply:
I have modified my pull request to display install error messages in a popup dialog.

Reply:
This looks good now, thanks.